core/templates/components/state-editor/state-interaction-editor/state-interaction-editor.directive.ts

```diff
@@ -367,6 +388,16 @@ angular.module('oppia').directive('stateInteractionEditor', [
                     };
 
                     $scope.save = function() {
+                      // If the interaction type is MultipleChoiceInput, we
+                      // need to handle the deletion of answer choices and
+                      // update the answer groups accordingly in order to match
+                      // the new answer choices.
+                      if (StateInteractionIdService.savedMemento ===
```

Comment:
Bit confused about why this is done here -- I don't see a corresponding thing for ItemSelection. Could you explain the disparity?

In general follow the existing conventions. (See what ItemSelection does.) We don't want two different approaches for two different interactions.

Reply:
In the case of `ItemSelection` interaction, The updating of answer choice is handled completely in `Responses.service.ts` based on the `oldAnswerChoices` and the `newAnswerChoices`.
But as we discussed earlier and decided to use the list-editor to keep track of the answer choices modifications (`$scope.oldToNewListMappingObject`) in the case of `MultipleChoiceInput`. Hence, we have to update the answer group here in the `state-interaction-editor.directive.ts` because `$scope.oldToNewListMappingObject` is not accessible from `Responses.service.ts`.
If we have to follow the approach used for handling `ItemSelection`, the `$scope.oldToNewListMappingObject` from the list editor becomes useless and we have to manually compare the `oldAnswerChoices` and `newAnswerChoices`(as in `ItemSelection`) and update the answers according to that.
Pls correct me If I'm missing something

Reply:
Ah, I see, that makes sense. Please leave clear detailed comments in the codebase to explain the divergence.

I still don't quite understand why you can't use Responses.service.ts though. Couldn't you pass the mapping object to it? That way, we keep all the interaction-type-specific logic in one place. (I.e. you don't need to follow exactly the approach for ItemSelection but at least all the code that goes "if interaction ID is equal to XXX" is in a single file.)

Reply:
Yeah I'll try passing the mapping object to `Responses.service.ts`